pkg/skaffold/schema/latest/config.go

```diff
@@ -851,6 +854,14 @@ type ArtifactType struct {
 	CustomArtifact *CustomArtifact `yaml:"custom,omitempty" yamltags:"oneOf=artifact"`
 }
 
+// ArtifactDependency describes a specific build dependency for an artifact.
+type ArtifactDependency struct {
+	// Alias is the alias for this image that can be referenced in the builder.
+	Alias string `yaml:"alias" yamltags:"required"`
+	// ImageName is the artifact image name.
+	ImageName string `yaml:"image" yamltags:"required"`
```

Comment:
hmm that's an interesting point. `Alias` was originally `Name` which we changed because that was confusing with `ImageName`. but can we reference modules by `ImageName`? if we don't allow the same image name to be used in two distinct artifacts then I don't see why not...

Reply:
I'm in favor of keeping them separate because `alias` is gonna mean an environment variable for docker builder and maybe others. As such there are different naming restrictions like no hyphens, etc. We probably don't want to add new restrictions to `ImageName` field.

Reply:
I've made `alias` field optional and default to the value of `image`. This way users can get away with just defining `image` when it'll work fine as an `alias` too.

Reply:
Also reordered to have `ImageName` first as it's the only required field